docs/source/modules/schedule.rst

```diff
@@ -1,10 +1,114 @@
+that fulfill the highest number of requests. In generating the schedule, the scheduler will ensure all request
+constraints are met and that there exist no conflicts between scheduled observations. The :class:`Scheduler` class
+documentation can be found below and an example usage can be found in the :ref:`Scheduling Workflow` tutorial.
+
+The :class:`Scheduler` class can be imported via the following:
+
+.. code-block:: python
+
+   from celest.schedule import Scheduler
+
+.. autoclass:: celest.schedule.scheduler.Scheduler
    :members:
-   :show-inheritance:
    :noindex:
+
+Scheduling Algorithm
+--------------------
+
+The scheduling algorithm used by the :class:`Scheduler` class is an adaptive large neighborhood search (ALNS)
+metaheuristic with a simulated annealing acceptance criterion and is adapted from the work of Xiaolu Liu, et al. (2017).
+[Liu+17]_ The following gives a brief overview to the algorithm which can be read in greater detail from the paper by
+Liu et al. [Liu+17]_
+
+.. [Liu+17] Xiaolu Liu et al. “An adaptive large neighborhood search metaheuristic for agile satellite scheduling with
+   time-dependent transition time”. en. In: Computers Operations Research 86 (Oct. 2017), pp. 41–53. issn: 03050548.
+   doi: 10.1016/j.cor.2017.04.006. url: https://linkinghub.elsevier.com/retrieve/pii/S0305054817300977.
+
+Initial Solution
+^^^^^^^^^^^^^^^^
+
+The scheduling algorithm uses the sum of scheduled requires priorities as a measure of the quality of the solution.
+That is, the goal of the algorithm is to determine the highest sum of priorities of all scheduled observations. As a
+result, a good initial solution is greedy one. The :class:`Scheduler`'s ALNS algorithm sorts all requests in decreasing
+order of their priorities and then in increasing order of start time; the scheduler then attempts to schedule all tasks
+in order whilst adhering to the request constraints and takes the resulting solution to be the initial solution.
+
+Note that if all requests are scheduled in the initial solution, the algorithm stops and returns the initial solution.
+
+Scheduler Iteration
+^^^^^^^^^^^^^^^^^^^
+
+The general methodology of a neighborhood search algorithm is to create a new solution in the neighborhood of the
+current solution by making small modifications through the use of destroy and repair operations. In each iteration,
+the scheduler will apply destroy and repair operation to the current solution to change it slightly. The cost of the
+resulting solution (the sum of the priorities of scheduled requests) is determined and compared to the cost of the
+current solution. The new solution is accepted by the :ref:`simulated annealing acceptance criterion<Acceptance Criterion>`.
+If the new solution is better than the best solution to date, the best solution is updated.
+
+Acceptance Criterion
+^^^^^^^^^^^^^^^^^^^^
+
+A new solution is accepted by the simulated annealing acceptance criterion. If the cost of the new solution is better
```

Comment:
There is a repetition here. Remove the repetition and combine the first two sentences.